### DISCUSSION TITLE
Ingester - write to WAL: no space left on device

### DISCUSSION THREAD

**Discussion by luvpreetsingh** [grafana/mimir#4808] [Help and support] [answered]
Hi,

I keep getting this error, 
```
level=warn ts=2023-04-20T19:52:06.420650274Z caller=grpc_logging.go:38 method=/cortex.Ingester/Push duration=3.952285ms err="user=prod-kube: write to WAL: log samples: write /data/tsdb/prod-kube/wal/00021776: no space left on device" msg=gRPC
```

That obviously means my ingesters are out of disk space. When I try to increase the disk space it doesn't allow me,

```
Error: UPGRADE FAILED: cannot patch "mimir-ingester" with kind StatefulSet: StatefulSet.apps "mimir-ingester" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden
```

If the size increase is forbidden, how can I solve this issue?

**Comment by luvpreetsingh** [comment]
I am using helm chart version 3.0.0

**Comment by aknuds1** [comment]
This is a Kubernetes specific issue and not really Mimir related, i.e. that you can't directly patch a [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) apart from the fields 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds'. I don't know the procedure for increasing the size of persistent volumes off the top of my head, maybe someone else will chime in?

**Reply by luvpreetsingh** [reply]
Someone on grafana slack did tell me that this is annoying.

Will this work? The way he said in the 2nd message.
![mimir-stateful-set](https://user-images.githubusercontent.com/23127804/233615427-c567f4b8-5070-4f98-8f35-ce34557f43a6.png)

**Comment by luvpreetsingh** [comment, marked as answer]
I solved by directly editing the pvc using kubectl. Thanks.
